src/rust-crypto/rust-crypto.ts

```diff
@@ -2138,6 +2151,34 @@ export class RustCrypto extends TypedEventEmitter<RustCryptoEvents, CryptoEventH
     public async getOwnIdentity(): Promise<RustSdkCryptoJs.OwnUserIdentity | undefined> {
         return await this.olmMachine.getIdentity(new RustSdkCryptoJs.UserId(this.userId));
     }
+    /**
+     * Handles the receipt of a to-device message, specifically for processing
+     * "io.element.msc4268.room_key_bundle" message types.
+     *
+     * @param payload - The received to-device message payload, which includes
+     * the message content and optional encryption information.
+     */
+    public async onReceiveToDeviceMessage(payload: ReceivedToDeviceMessage): Promise<void> {
+        if (payload.message.type != "io.element.msc4268.room_key_bundle") {
+            return;
+        }
+
+        const { message, encryptionInfo } = payload;
+        const claimedSender = encryptionInfo?.sender ?? message.sender;
+
+        // Validate room ID
+        const roomId = message.content.room_id;
+        if (typeof roomId !== "string") {
+            return;
+        }
+
+        // Check if the room is in the map of rooms we expect to receive bundles from, otherwise discard.
+        if (this.roomsPendingKeyBundles[roomId] !== claimedSender) {
+            return;
+        }
+
+        await this.maybeAcceptKeyBundle(roomId, claimedSender);
```

Comment:
we need to remove the entry from `roomsPendingKeyBundles` if this is successful?

Reply:
Fixed in 62a2af44235ef9f45b1c9cd409e87468d48d9b43